tls/s2n_config.c

```diff
@@ -479,6 +485,52 @@ int s2n_config_add_cert_chain_and_key_to_store(struct s2n_config *config, struct
     *to_insert = cert_key_pair;
     GUARD(s2n_config_build_domain_name_to_cert_map(config->domain_name_to_cert_map, cert_key_pair));
 
+    if (!config->default_certs_are_explicit) {
```

Comment:
Does this mean that you have to call `s2n_config_set_cert_chain_and_key_defaults` before `s2n_config_add_cert_chain_and_key_to_store` (to set default_certs_are_explicit)? Or could you call them in either order and end up with the same default certs (since `s2n_config_set_cert_chain_and_key_defaults` overwrites previously set values)?

Checking my understanding: If you call `s2n_config_and_pem_to_trust_store` with an RSA cert first, and then `s2n_config_set_cert_chain_and_key_defaults` with an ECC cert second, there will only end up being an RSA default since the API clears out all of default_cert_per_auth_method?

Reply:
Ok, couple different questions here. Firstly, you can call them in either order. When you call the API to set the default, this is taken as an explicit signal that you only want these certs to be considered defaults by s2n. Before you call this API to set defaults, you can add certs/keys to the s2n config store like normal and s2n will attempt to auto-choose defaults based on the ordering (this is the current behavior and I wanted to preserve this). Your last question is a bit wrong, but I think it was a typo: If you add an RSA cert first then set an ECDSA cert as default, then only the ECDSA cert will be used as a default cert because that's what you explicitly told s2n to do using the API.

Reply:
Yep, that was a typo.